dracut-saltboot/saltboot/saltboot.sh

```diff
@@ -88,16 +97,16 @@ if [ -z "$HAVE_MINION_ID" ] ; then
         UNIQUE_SUFFIX="-${MACHINE_ID:0:4}"
     fi
 
-    if [ -n "$HOSTNAME" -a -z "$DISABLE_HOSTNAME_ID" ] ; then
+    if [ -n "$FQDN" -a -z "$DISABLE_HOSTNAME_ID" ] ; then
         # MINION_ID_PREFIX can be specified on kernel cmdline
         if [ -n "$MINION_ID_PREFIX" ] ; then
-            echo "$MINION_ID_PREFIX.$HOSTNAME$UNIQUE_SUFFIX" > /etc/salt/minion_id
+            echo "$MINION_ID_PREFIX:$FQDN$UNIQUE_SUFFIX" > /etc/salt/minion_id
         else
```

Comment:
The idea behind the original code was that MINION_ID_PREFIX is a shorter alternative to domain name. Using both is IMHO redundant.
Also, if we change this, we have to update documentation.

Reply:
The documentation already states FQDN in the Template. But I agree that this can be considered API change. I will revert this part and create new issue to discuss various other possibilities how to generate naming of the terminal.